### ISSUE TITLE
[iOS] Review and reword permission usage text

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#4139]
While reviewing #4089 before merge, I noticed (https://github.com/zulip/zulip-mobile/pull/4089#issuecomment-639155183 ) that:
* if you go to download an image from the lightbox,
* and you haven't already given the app permissions on the photo library,
* we show (well, cause the system to show) a permission request with the text

  > **"Zulip" Would Like to Access Your Photos**
  > Upload photos to a Zulip realm

That text can be improved in at least a couple of ways:
* We're not trying to upload at this point! Rather we're trying to *download*.

  The text should match what we're actually trying to do; that's an important element of maintaining users' trust.

* The phrase doesn't feel like it fits, grammatically, in the context.

* The term "realm" generally isn't in our user-facing messages; it makes more sense for Zulip developers than for most users. A more natural phrasing (modulo the other issues mentioned) would be simply "Upload photos to Zulip".

Looking in `ios/ZulipMobile/Info.plist`, we have a handful of these strings (other than the boilerplate for those we never use; see c8d0c8df5 for why those are there):
```
       <key>NSCameraUsageDescription</key>
       <string>Take a photo and upload it to a Zulip realm</string>
       <key>NSPhotoLibraryAddUsageDescription</key>
       <string>Download photos from Zulip realm</string>
       <key>NSPhotoLibraryUsageDescription</key>
       <string>Upload photos to a Zulip realm</string>
```
So it seems like for the first problem above, we actually have a more appropriate string, and there's just some debugging to do to see why that one doesn't get used. (One of these sounds more general than the other; maybe the more general one takes precedence?)

The other two problems are quite parallel across all three, and whatever rewording we do we can apply to all three in parallel.

**Comment by Gautam-Arora24** [comment]
I would like to work on this issue

**Comment by gnprice** [comment]
We recently noticed these again; we should really fix them. Chat thread here:
https://chat.zulip.org/#narrow/stream/48-mobile/topic/permission.20usage.20text/near/1391574
and I think we have some pretty reasonable candidate new strings.

**Comment by gnprice** [comment]
As written, this issue is really two pretty separate tasks:
* Debug why, when saving an image from the lightbox, we request general photo-library permission instead of only permission to add to it.
* Write better text for all three of these permission requests.

Let's focus this issue thread about the latter. I'll file a separate issue for the former.
